jamventProto/jamctrllib/JamCtrlMgrInterface.h

```diff
@@ -0,0 +1,181 @@
+#ifndef __JAM_CTRL_MGR_INTERFACE_H__
+#define __JAM_CTRL_MGR_INTERFACE_H__
+
+/**
+ * @class JamCtrlMgrInteface
+ * @brief Interface class for the control manager
+ */
+class JamCtrlMgrInterface
+{
+  public:
+    using IntegerNumType = uint32_t; ///< Data type for input values
+    using FloatNumType = double;   ///< Data type for graph values
+
+    static constexpr FloatNumType BarPerCmh2o = 0.000980665;       ///< 0.000980665 Bar / 1 cmH2O
+    static constexpr FloatNumType Cmh2oPerBar = 1.0 / BarPerCmh2o; ///< 1019.716 cmH2O / 1 Bar
+
+    // Data in from the system.
+    /**
+    * @struct DataIn
+    * @brief Data from system
+    */
+    struct DataIn
+    {
+        /// User input settings
+
+        /// Oxygen to air mix
+        /// Units: Percentage (%)
+        /// Range: 21 to 100
+        IntegerNumType fiO2_pc;
+
+        /// Tidal volume
+        /// Units: Milliliters
+        /// Range: 10 to 1500
+        IntegerNumType tidalVol_ml;
+
+        /// Respiratory rate
+        /// Units: None. Breaths per minute
+        /// Range: 1 to 40
+        IntegerNumType respRate;
+
+        /// Inhale/Exhale ratio.
+        /// Units: None. Choice of ratios
+        /// Range: 0 (1:4) to 6 4:1)
+        ///   0 = 1:4
+        ///   1 = 1:3
+        ///   2 = 1:2
+        ///   3 = 1:1
+        ///   4 = 2:1
+        ///   5 = 3:1
+        ///   6 = 4:1
+        IntegerNumType ieRatio;
+
+        /// Positive end expiratory pressure
+        /// Units: Bar
+        /// Range: 0.0 to 0.0196133 Bar (internal units), 0.0 to 20.0 cmH2O (display units)
+        FloatNumType peep_bar;
+
+        // Valve states.
+        bool valveAO2Open; ///< True when valve A is open. False when closed.
+        bool valveBAirOpen; ///< True when valve B is open. False when closed.
+        bool valveCInhaleOpen; ///< True when valve C is open. False when closed.
+        bool valveDExhaleOpen; ///< True when valve D is open. False when closed.
+
+        // Graph data
+        /// Reservior pressure (high)
+        /// Units: Bar
+        /// Range: 0.0 Bar to 5.0 Bar
+        FloatNumType pRes_bar; // Pressure in reservior
+
+        /// System pressure (low)
+        /// Units: Bar
+        /// Range: 0.01 Bar to 1.2 Bar
+        FloatNumType pSys_bar; // Pressure at the patient
+
+        /// O2 percentage in mix
+        /// Units: Percentage (%)
+        /// Range: 21.0 to 100.0
+        FloatNumType o2_pc;
```

Comment:
The gas blending math requires O2 mix percentage as a fractional decimal, e.g. 21.0% as 0.210. This is something the gas mixing class could do by just dividing by 100, but maybe we want an official get/conversion function in this interface so that all conversions are done in one location?

Reply:
Good you pointed this out.  I didn't even think of it.  So maybe the range should be 0.21 to 1.0.  @dcstraney , what range will the hw layer return?

Reply:
Hardware layer returns 0-100 (as a float); will change that to keep it consistent everywhere